sqlx-core/src/net/socket/mod.rs

```diff
@@ -182,10 +183,18 @@ impl<S: Socket + ?Sized> Socket for Box<S> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct TcpKeepalive {
+    pub time: Duration,
+    pub interval: Duration,
+    pub retries: u32,
+}
+
 pub async fn connect_tcp<Ws: WithSocket>(
     host: &str,
     port: u16,
     with_socket: Ws,
+    keepalive: &Option<TcpKeepalive>,
```

Comment:
```suggestion
    keepalive: Option<&TcpKeepalive>,
```

Would this not be a better API?
https://users.rust-lang.org/t/api-design-option-t-vs-option-t/34139/2

Given that `socket2::TcpKeepalive` is `Copy`, I think copying that derive and dropping the reference might be more ergonomic. What do you think?

Reply:
I didn't notice that `TcpKeepalive` in `socket2` is `Copy`.
I will adjust it. Thx.

Reply:
I didn't notice the API design preference in Rust. Thank you for pointing this out for me!